waiter/src/waiter/async_request.clj

```diff
@@ -139,8 +139,9 @@
    The function assumes location begins with a slash.
    This method wires up the completion and status check callbacks for the monitoring system.
    It also modifies the status check endpoint in the response header."
-  [router-id async-request-store-atom make-http-request-fn instance-rpc-chan service-id metric-group {:keys [host port] :as instance}
-   {:keys [request-id] :as reason-map} request-properties location response-headers]
+  [router-id async-request-store-atom make-http-request-fn instance-rpc-chan response
+   service-id metric-group {:keys [host port] :as instance}
+   {:keys [request-id] :as reason-map} request-properties location]
```

Comment:
nitpick: `response` could be the last argument in the function as this is like a response handler

Reply:
`response` is first effectively, as we end up using this with `partial`, which works nicely with `->`.  Which response handlers have it at the end?

Reply:
Request handlers tend to have it at the end that's why the nitpick to have response at the end for response handlers. Also, this is actually not even the response :)